### PR TITLE
Module3

### PR DESCRIPTION
**Pull request by DanilaSamusev**

### CODE REVIEW

File: Module3/Module3_4/Program.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        
+        static int GetNumber()
+        {
+            while (true)
+            {
+                Console.Write("Enter the number: ");
+
+                if (int.TryParse(Console.ReadLine(), out int parsedNumber))
+                {
+                    return parsedNumber;
```
Comment (az00s) [inline Module3/Module3_4/Program.cs:22]: good variable name

File: Module3/Module3_4/Program.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        {
+            string strNumber = number.ToString();
+
+            if (number < 0)
+            {
+                strNumber = strNumber.Remove(0, 1);
+            }
+            
+            char[] numberArray = strNumber.ToCharArray();
+            Array.Reverse(numberArray);
```
Comment (az00s) [inline Module3/Module3_4/Program.cs:39]: bad idea => the target of this task is to check how you can do it without existing .Net methods. you should have used loops.

File: Module3/Module3_5/Program.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+namespace Module3_5
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            int number = GetNumber();
+            int digit = GetDigit();
+            int reformedNumber = int.Parse(number.ToString().Replace($"{digit}", ""));
```
Comment (az00s) [inline Module3/Module3_5/Program.cs:11]: remove $"{...}"

File: Module3/Module3_4/Program.cs (hunk 2 above)
Comment (DanilaSamusev) [reply Module3/Module3_4/Program.cs:39]: That's why I've received 5?

File: Module3/Module3_5/Program.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+                Console.WriteLine("Error: input data has to be a positive integer!\n-------------------------");
+            }
+        }
+
+        static int GetDigit()
+        {
+            while (true)
+            {
+                Console.Write("Inter the the digit: ");
```
Comment (az00s) [inline Module3/Module3_5/Program.cs:34]: Enter

File: Module3/Module3_5/Program.cs [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+                if (int.TryParse(Console.ReadLine(), out int parsedDigit) 
+                    && parsedDigit < 10 && parsedDigit >= 0)
+                {
+                    return parsedDigit;
+                }
+
+                Console.WriteLine("Error: input data has to be a digit!\n-------------------------");
```
Comment (az00s) [inline Module3/Module3_5/Program.cs:42]: if you expect digit which is less 10 then use appropriate message

File: Module3/Module3_5/Program.cs (hunk 3 above)
Comment (az00s) [inline Module3/Module3_5/Program.cs:11]: why you  Parse it => use string

File: Module3/Module3_5/Program.cs [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        }
+
+        static int GetDigit()
+        {
+            while (true)
+            {
+                Console.Write("Inter the the digit: ");
+
+                if (int.TryParse(Console.ReadLine(), out int parsedDigit) 
+                    && parsedDigit < 10 && parsedDigit >= 0)
```
Comment (az00s) [inline Module3/Module3_5/Program.cs:37]: what about
 parsedDigit/10 == 0

File: Module3/Module3_6/Module3_6/ArrayInitializer.cs [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    public class ArrayInitializer
+    {
+        public double[] FillArray()
+        {
+            int arraySize = GetArraySize();
+            double[] array = new double[arraySize];
+
+            for (int i = 0; i < arraySize;)
+            {
+                Console.Write($"Inter the {i + 1} array number: ");
```
Comment (az00s) [inline Module3/Module3_6/Module3_6/ArrayInitializer.cs:14]: Enter

File: Module3/Module3_6/Module3_6/Program.cs [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+            }
+        }
+
+        static void PrintArray(double[] array, string message)
+        {
+            Console.Write($"{message} array is:");
+
+            foreach (var number in array)
+            {
+                Console.Write($" {number}");
```
Comment (az00s) [inline Module3/Module3_6/Module3_6/Program.cs:30]: just number instead of $" {number}"

File: Module3/Module3_7/Program.cs [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+                if (array[i] > array[i - 1])
+                {
+                    numbersList.Add(array[i]);
+                }
+            }
+
+            return numbersList;
+        }
+
+        private static void PrintIEnumerable<T>(T array, string message) where T : IEnumerable
```
Comment (az00s) [inline Module3/Module3_7/Program.cs:41]: generic is not needed here

File: Module3/Module3_8/Program.cs [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+                
+                if (Function(rightBound) * Function(solution) < 0)
+                {
+                    leftBound = solution;
+                }
+            }
+            
+            return (leftBound + rightBound) / 2;
+        }
+        static double Function(double x)
```
Comment (az00s) [inline Module3/Module3_8/Program.cs:56]: bad name. use CalculateSomething, MeasureSomething or GetSomething

**Pull request merged**